src/cpu/aarch64/acl_convolution_utils.cpp

```diff
@@ -193,11 +194,96 @@ status_t init_conf(acl_conv_gemm_conf_t &acp, memory_desc_t &src_md,
+
+    // clang-format off
+    // Validate convolution manually to check for return status
+    arm_compute::NEGEMMConvolutionLayer acl_gemm_conv;
+    auto acl_st = acl_gemm_conv.validate(
+        &acp.src_info,
+        &acp.wei_info,
+        acp.with_bias ? &acp.bia_info : nullptr,
+        &acp.dst_info,
+        acp.padstride_info,
+        acp.weights_info,
+        acp.dilation_info,
+        acp.act_info);
+    // clang-format on
+    if (acl_st.error_code() != arm_compute::ErrorCode::OK) {
+        return status::unimplemented;
+    }
+
+    return st;
+}
+
+status_t init_conf_wino(acl_conv_conf_t &acp, memory_desc_t &src_md,
+        memory_desc_t &weights_md, memory_desc_t &dst_md,
+        memory_desc_t &bias_md, const convolution_desc_t &cd,
+        const primitive_attr_t &attr) {
+
+    // General Compute Library checks, memory tags are also set there
+    status_t st
+            = acl_init_conf(acp, src_md, weights_md, dst_md, bias_md, cd, attr);
+
+    const bool wino_shape_ok // unit strides only, no dilations
+            = (acp.padstride_info.stride() == std::pair<uint, uint> {1, 1})
+            && (acp.dilation_info == arm_compute::Size2D(1, 1));
+    if (!wino_shape_ok) return status::unimplemented;
+
+    // clang-format off
+    // Validate convolution manually to check for return status
+    arm_compute::NEWinogradConvolutionLayer acl_wino_conv;
+    auto acl_st = acl_wino_conv.validate(
+        &acp.src_info,
+        &acp.wei_info,
+        acp.with_bias ? &acp.bia_info : nullptr,
+        &acp.dst_info,
+        acp.padstride_info,
+        acp.act_info,
+        true); // enable_fast_math flag in ACL Winograd
+    // clang-format on
+    if (acl_st.error_code() != arm_compute::ErrorCode::OK) {
+        return status::unimplemented;
+    }
+
+    const unsigned int width_idx
+            = get_data_layout_dimension_index(acp.src_info.data_layout(),
+                    arm_compute::DataLayoutDimension::WIDTH);
+    const unsigned int height_idx
+            = get_data_layout_dimension_index(acp.src_info.data_layout(),
+                    arm_compute::DataLayoutDimension::HEIGHT);
+    const unsigned int src_ch_idx
+            = get_data_layout_dimension_index(acp.src_info.data_layout(),
+                    arm_compute::DataLayoutDimension::CHANNEL);
+    const unsigned int dst_ch_idx
+            = get_data_layout_dimension_index(acp.dst_info.data_layout(),
+                    arm_compute::DataLayoutDimension::CHANNEL);
+    // Under these conditions, fallback to faster GEMM-based convolution
+    // unless the user explicitly specifies Winograd algorithm
+    if (one_of(true, acp.src_info.dimension(height_idx) > 112,
+                acp.src_info.dimension(width_idx) > 112,
+                acp.src_info.dimension(src_ch_idx) < 64,
+                acp.dst_info.dimension(dst_ch_idx) < 64,
+                dnnl_get_max_threads() > 28)
+            && cd.alg_kind != alg_kind::convolution_winograd) {
```

Comment:
nit:

```suggestion
            && cd.alg_kind == alg_kind::convolution_auto) {
```

mainly, for clarity (mentioned in the comment above as well).